lib/log.c

```diff
@@ -212,6 +217,12 @@ void vzlog(int priority, const char *format, va_list args)
 	struct timestamp_control tsctl;
 	tsctl.already_rendered = 0;
 	struct zlog *zl = zlog_default;
+	va_list ac;
+
+	/* call external hook */
+	va_copy(ac, args);
+	hook_call(zebra_ext_log, priority, format, ac);
+	va_end(ac);
 
```

Comment:
is it possible to avoid va_copy ?

Reply:
@pguibert6WIND please see my comments in the "Summary" section of the PR, above, both regarding the `va_copy` and the use case. In case someone has registered a callback we do need to copy the arguments, as I understand they might be altered by the callback otherwise. 
Someone more familiar with FRR hooks than me can maybe tell me if there is a way to check at runtime whether a hook was registered, to avoid copying arguments when it's not needed. 

Alternatively we could go with the other implementation, which allows you to explicitly register an external callback with a zlog API - I switched to hooks because of a comment from David in the slack channel when I asked for feedback on that other commit, but either way works for me and I'm happy to modify them as required.

Reply:
Thanks Emanuele. I think it will be ok, if that second pair of eyes is ok.
I am not an expert in hooks, but I see it as a bump that may shift some extra cpu consumption.

Reply:
There can be more than one callback, so it's really the callback that must do the va_copy since we'd need to make a separate copy per callback...

Reply:
You are right, fixed that